### ISSUE TITLE
Duplicate videos in background player causes it to behave improperly

### ISSUE THREAD

**Issue by luckcolors** [TeamNewPipe/NewPipe#6122]
### Checklist

- [x] I am using the latest version - 0.21.1 
- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one bug. I will open one issue for every bug report I want to file.

### Steps to reproduce the bug
1. Equeue to the bakground player 1 video, let's call it video X
2. Equeue to the bakground player 1 video, let's call it video Y
3. Enque to the background player video X again.
   Now our playlist looks like this:
   - X
   - Y
   - X
   Note this specific playlist pattern is not the cause as this problem happens when any kind of duplicate is present in any order, for example to test this it's even enough to just have one video present more than one time for example.
   - X
   - X
4. Start playing from the beginning is fine but the problem happens on starting to play the first duplicated video

### Actual behaviour
If the user selects to start playing on the first appearance of a any video, everything works fine the playlist plays until reaching a duplicate video on wich it starts loading forever and gets stuck on.
If the user manually selects to play from a duplicated video the player is unable to do so and "jumps" to the first appearance in the playlist of said video.

### Expected behavior
One would either expect to not be able to add duplicate videos in any playlist or for them to act as an individual entry in the playlist.
This is especially a problem when enqueuing multiple playlists or simply re-enquequing an already present video as it can cause duplicates to appear wich then proceed to cause this issue.

If i were to be personally in favour of one of those 2 options i would assume allowing for duplicates is the most natural option.

**Comment by ShinyLuxray** [comment]
I'll add that this issue is present in version 0.21.2, and affects videos in foreground mode as well.

Anecdotally, this bug must have been introduced at some point as I first encountered it today when I played a saved playlist containing duplicate videos which worked before. That is to say, I do appreciate contributors' efforts as the queuing interface has improved substantially these last few months.
